File: packages/evals/tasks/bench/act/namespace_xpath.ts
import { defineBenchTask } from "../../../framework/defineTask.js";

export default defineBenchTask(
  { name: "namespace_xpath" },
  async ({ debugUrl, sessionUrl, stagehand, page, logger }) => {
    try {
      await page.goto("https://browserbase.github.io/stagehand-eval-sites/sites/namespaced-xpath/");

      await stagehand.act("fill 'nunya' into the 'type here' form");

      const inputValue = await page.locator("#ns-text").inputValue();
      // confirm that the form was filled
      const formHasBeenFilled = inputValue === "nunya";

      return {
        _success: formHasBeenFilled,
        debugUrl,
        sessionUrl,
        logs: logger.getLogs(),
      };
    } catch (error) {
      return {
        _success: false,
        error: error instanceof Error ? error.message : String(error),
        debugUrl,
        sessionUrl,
        logs: logger.getLogs(),
      };
    }
  },
);

File: packages/sdk-ruby/evals/tasks/act/namespace_xpath.rb
# frozen_string_literal: true

# Port of packages/evals/tasks/bench/act/namespace_xpath.ts

Evals.define_task("namespace_xpath") do |t|
  t.page.goto("https://browserbase.github.io/stagehand-eval-sites/sites/namespaced-xpath/")

  t.stagehand.act("fill 'nunya' into the 'type here' form")

  input_value = t.page.locator("#ns-text").input_value
  # confirm that the form was filled
  form_has_been_filled = input_value == "nunya"

  { _success: form_has_been_filled }
end
